types/rlp_unmarshal.go

```diff
@@ -361,11 +361,23 @@ func (t *Transaction) UnmarshalRLPFrom(p *fastrlp.Parser, v *fastrlp.Value) erro
 		return err
 	}
 
-	// Type
 	if len(elems) >= 10 {
+		// Type
 		if t.Type, err = ReadRlpTxType(elems[9]); err != nil {
 			return err
 		}
+
+		if t.IsStateTx() {
+			// From
```

Comment:
Can you extend the comment with more info on why we're doing this address assignment?

Reply:
Fixed [97f3d78](https://github.com/0xPolygon/polygon-edge/pull/796/commits/97f3d78fdf9f276162c65df6b07b1d855b190e3d)